modules/dataframeqt/src/processors/dataframedocktable.cpp

```diff
@@ -0,0 +1,288 @@
+#include <inviwo/dataframeqt/dataframetableview.h>
+
+#include <functional>  // for __base
+#include <limits>      // for numeric_limits
+#include <map>         // for map
+#include <utility>     // for move
+#include <memory>      // for shared_ptr, unique_ptr
+
+#include <glm/vec2.hpp>  // for operator!=
+#include <QObject>       // for Q_OBJECT
+#include <QVBoxLayout>
+#include <QMainWindow>
+
+namespace inviwo {
+
+namespace {
+QMainWindow* findParent(Processor* p) {
+    auto* net = p->getNetwork();
+
+    std::vector<std::pair<Processor*, QMainWindow*>> candidates;
+    net->forEachProcessor([&](Processor* processor) {
+        auto* widget = processor->getProcessorWidget();
+        if (auto* mw = dynamic_cast<QMainWindow*>(widget)) {
+            candidates.emplace_back(processor, mw);
+        }
+    });
+
+    if (!candidates.empty()) {
+        return candidates.front().second;
+    }
+
+    return utilqt::getApplicationMainWindow();
+}
+
+}  // namespace
+
+DataFrameDockTableWidget::DataFrameDockTableWidget(Processor* p)
+    : ProcessorDockWidgetQt(p, utilqt::toQString(p->getDisplayName()), findParent(p))
+    , tableview_{new DataFrameTableView(this)} {
+
+    setAllowedAreas(Qt::AllDockWidgetAreas);
+    auto* mw = findParent(p);
+    mw->addDockWidget(Qt::LeftDockWidgetArea, this);
+
+    tableview_->setMouseTracking(true);
+    tableview_->setAttribute(Qt::WA_OpaquePaintEvent);
+
+    QSizePolicy sp(QSizePolicy::MinimumExpanding, QSizePolicy::MinimumExpanding);
+    sp.setVerticalStretch(5);
+    sp.setHorizontalStretch(5);
+    tableview_->setSizePolicy(sp);
+
+    setFocusProxy(tableview_);
+
+    auto* mainWidget = new QWidget();
+    auto* layout = new QVBoxLayout(mainWidget);
+    layout->setAlignment(Qt::AlignTop);
+    const auto space = utilqt::refSpacePx(this);
+    layout->setContentsMargins(0, space, 0, space);
+    layout->setSpacing(space);
+
+    layout->addWidget(tableview_);
+
+    setWidget(mainWidget);
+}
+
+DataFrameDockTableWidget::~DataFrameDockTableWidget() = default;
+
+void DataFrameDockTableWidget::setManager(BrushingAndLinkingManager& manager) {
+    tableview_->setManager(manager);
+}
+void DataFrameDockTableWidget::setDataFrame(std::shared_ptr<const DataFrame> dataframe,
+                                            bool categoryIndices) {
+    tableview_->setDataFrame(std::move(dataframe), categoryIndices);
+}
+void DataFrameDockTableWidget::setIndexColumnVisible(bool visible) {
+    tableview_->setIndexColumnVisible(visible);
+}
+void DataFrameDockTableWidget::setFilteredRowsVisible(bool visible) {
+    tableview_->setFilteredRowsVisible(visible);
+}
+void DataFrameDockTableWidget::brushingUpdate() { tableview_->brushingUpdate(); }
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo DataFrameDockTable::processorInfo_{
+    "org.inviwo.DataFrameDockTable",  // Class identifier
+    "Data Frame Dock Table",          // Display name
```

Comment:
```suggestion
    "DataFrame Dock Table",          // Display name
```

Reply:
Not sure if I like this, should be either Data Frame for Dataframe... but I'll change it for now.